test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/DefaultStorageCredentialsValidatorTests.cs

```diff
@@ -15,6 +16,45 @@ namespace Microsoft.Azure.WebJobs.Host.UnitTests.Executors
 {
     public class DefaultStorageCredentialsValidatorTests
     {
+        [Fact]
+        public void UseReflectionToAccessIsDevStoreAccountFromCloudStorageAccount()
+        {
+            var isDevStoreAccountProperty = typeof(CloudStorageAccount).GetProperty("IsDevStoreAccount", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.NotNull(isDevStoreAccountProperty);
+        }
+
+        [Fact]
+        public void StorageAccount_IsDevStoreAccount_StorageEmulatorRunning()
+        {
+            var validator = new DefaultStorageCredentialsValidator();
+
+            var storageMock = new Mock<IStorageAccount>(MockBehavior.Strict);
+            var blobClientMock = new Mock<IStorageBlobClient>();
+            var queueClientMock = new Mock<IStorageQueueClient>();
+            var queueMock = new Mock<IStorageQueue>();
+
+            storageMock.Setup(s => s.Credentials)
+                .Returns(new StorageCredentials("name", string.Empty));
+
+            storageMock.Setup(s => s.CreateBlobClient(null))
+                .Returns(blobClientMock.Object)
+                .Verifiable();
+
+            storageMock.Setup(s => s.SdkObject)
+                .Returns(() => {
+                    var account = new CloudStorageAccount(new StorageCredentials("name", string.Empty), false);
+                    typeof(CloudStorageAccount).GetProperty("IsDevStoreAccount", BindingFlags.NonPublic | BindingFlags.Instance).SetValue(account, true);
```

Comment:
Instead of updating this property, can you use actual emulator storage credentials? That'd make sure that it works e2e.